File: xNet/src/main/java/org/sqlunet/browser/PositionViewModel.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.sqlunet.browser;

import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.SavedStateHandle;
import androidx.lifecycle.ViewModel;

public class PositionViewModel extends ViewModel
{
	private static final String TAG = "PositionViewModel";

	private final SavedStateHandle savedStateHandle;

	public PositionViewModel(SavedStateHandle savedStateHandle)
	{
		this.savedStateHandle = savedStateHandle;
	}

	public void setPosition(int position)
	{
		Log.d(TAG, "setPosition " + position);
		this.savedStateHandle.set("position", position);
	}

	@NonNull
	public LiveData<Integer> getPositionLiveData()
	{
		return savedStateHandle.getLiveData("position");
	}
}

File: xNet/src/main/java/org/sqlunet/browser/PositionViewModel.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.browser

import androidx.lifecycle.LiveData
import androidx.lifecycle.SavedStateHandle
import androidx.lifecycle.ViewModel

class PositionViewModel(private val savedStateHandle: SavedStateHandle) : ViewModel() {

    fun setPosition(position: Int) {
        savedStateHandle["position"] = position
    }

    val positionLiveData: LiveData<Int>
        get() = savedStateHandle.getLiveData("position")
}
